common/types.go

```diff
@@ -475,3 +475,16 @@ func (d *Decimal) UnmarshalJSON(input []byte) error {
 		return err
 	}
 }
+
+// IsNil uses reflect to distinguish data that is truly nil.
```

Comment:
Is there any case where the data isn't truly nil? Just to know why this function was required.

Reply:
Some UTs send a `(evm.ChainContext)(nil)` to `eip4844.CalcBlobFee(*header.ExcessBlobGas, config)`, it will cause panic, if we just use `config == nil` to check nil.